ttnn/CMakeLists.txt

```diff
@@ -624,6 +623,12 @@ target_compile_options(
         -fno-var-tracking
 )
 
+if(WITH_PYTHON_BINDINGS)
+    target_compile_definitions(ttnn PUBLIC TTNN_WITH_PYTHON_BINDINGS=1)
+else()
+    target_compile_definitions(ttnn PUBLIC TTNN_WITH_PYTHON_BINDINGS=0)
```

Comment:
Are we concerned about our future selves (or others) using `#ifdef` instead of `#if`?  If we omit this one and only have the positive path, then both `#if` and `#ifdef` would give us the desired result in code.

Reply:
I typically prefer to use `#if` for our own defines, because if there's a typo in define name `#if` would give you an error